cli_tools/common/image/importer/api_inflater.go

```diff
@@ -47,22 +47,24 @@ func isCausedByAlphaAPIAccess(err error) bool {
 
 // apiInflater implements `importer.inflater` using the Compute Engine API
 type apiInflater struct {
-	request         ImageImportRequest
-	computeClient   daisyCompute.Client
-	storageClient   domain.StorageClientInterface
-	guestOsFeatures []*compute.GuestOsFeature
-	wg              sync.WaitGroup
-	cancelChan      chan string
-	logger          logging.Logger
+	request          ImageImportRequest
+	computeClient    daisyCompute.Client
+	storageClient    domain.StorageClientInterface
+	guestOsFeatures  []*compute.GuestOsFeature
+	wg               sync.WaitGroup
+	cancelChan       chan string
+	logger           logging.Logger
+	isShadowInflater bool
 }
 
-func createAPIInflater(request ImageImportRequest, computeClient daisyCompute.Client, storageClient domain.StorageClientInterface, logger logging.Logger) Inflater {
+func createAPIInflater(request ImageImportRequest, computeClient daisyCompute.Client, storageClient domain.StorageClientInterface, logger logging.Logger, isShadowInflater bool) *apiInflater {
 	inflater := apiInflater{
-		request:       request,
-		computeClient: computeClient,
-		storageClient: storageClient,
-		cancelChan:    make(chan string, 1),
-		logger:        logger,
+		request:          request,
+		computeClient:    computeClient,
+		storageClient:    storageClient,
+		cancelChan:       make(chan string, 1),
+		logger:           logger,
+		isShadowInflater: isShadowInflater,
```

Comment:
`calculateChecksum`?  (AFAICT that's the only difference between the two options)

Reply:
calculateChecksum is the internal logic which doesn't have to be exposed to the callers. There will be different behaviors based on whwther it's a shadow inflater. So it's a better abstraction to me.
Looking into the details, there are also some other differences - signals and err msgs.

Reply:
makes sense